include/time_series/multiprocess_time_series.hpp

```diff
@@ -87,20 +91,71 @@ class MultiprocessTimeSeries
             // sharing the max_length in the shared memory
             // (follower can query size for proper construction)
             shared_memory::set<size_t>(segment_id, "max_length", max_length);
+            shared_memory::set<Index>(
+                segment_id, "start_timeindex", start_timeindex);
         }
     }
 
+    MultiprocessTimeSeries(MultiprocessTimeSeries<T>&& other) noexcept
+        : internal::TimeSeriesBase<internal::MultiProcesses, T>(
+              std::forward<MultiprocessTimeSeries<T>>(other)),
+          indexes_(other.indexes_)
+    {
+    }
+
     /**
      * returns the max length used by a leading MultiprocessTimeSeries
      * of the corresponding segment_id
      */
     static size_t get_max_length(const std::string& segment_id)
     {
         size_t s;
+        shared_memory::get<size_t>(segment_id, "max_length", s);
+        return s;
+    }
+
+    /**
+     * returns the start index used by a leading MultiprocessTimeSeries
+     * of the corresponding segment_id
+     */
+    static Index get_start_timeindex(const std::string& segment_id)
+    {
+        Index index;
+        shared_memory::get<Index>(segment_id, "start_timeindex", index);
+        return index;
+    }
+
+    static MultiprocessTimeSeries<T> create_leader(std::string segment_id,
+                                                   size_t max_length,
+                                                   Index start_timeindex = 0)
+    {
+        bool leader = true;
+        return MultiprocessTimeSeries<T>(
```

Comment:
Isn't this using the constructor which is now marked deprecated?

Reply:
If it is the case, we would always get the warning, even when using it correctly.

Reply:
Indeed, things are not ideal now. Idea that once ready, we make the constructor protected, and remove the deprecated tag. I marked it deprecated mostly so that we do not forget to make it protected later on (could have made an issue instead)

Reply:
:+1: